Sources/PackageLoading/ManifestLoader.swift

```diff
@@ -442,29 +442,26 @@ public final class ManifestLoader: ManifestLoaderProtocol {
         completion: @escaping (Result<EvaluationResult, Error>) -> Void
     ) throws {
         do {
-            if localFileSystem.isFile(manifestPath) {
+            try withTemporaryDirectory { tempDir, cleanupTempDir in
+                let manifestTempFilePath = tempDir.appending(component: "manifest.swift")
+                try localFileSystem.writeFileContents(manifestTempFilePath, bytes: ByteString(manifestContents))
```

Comment:
The change looks good in that it gets rid of the special-casing.  Just so I understand, it does change so that now we always write the temporary file, whether or not we already have an original source file?

Reply:
That's right, for two reasons:
- this could lead to reading from the wrong file, because we were always reading from the local FS, no matter which one we were actually using to get the contents of the manifest (this originally sparked this PR)
- I'd like us to gain the ability to add to the manifest (e.g. by adding imports), this will allow things like making our non-implementation-only import of Foundation conditional on older tools-versions

Reply:
Absolutely, I was thinking along exactly the same lines for bullet two below.  Also for automatically getting struct definitions from things like plugins.  As long as the VFS option is solid in terms of the diagnostics (which the tests should show), then I think this is great.

Reply:
Yah, diagnostics look solid and also other context, like the delegate methods etc. We're now always passing down the original path to the loading machinery and the only thing that knows about the "real" path is the VFS.